core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -150,8 +153,17 @@ public Snapshot snapshot(SnapshotId snapshotId) {
         Repository repository = repositoriesService.repository(repositoryName);
         List<SnapshotId> snapshotIds = repository.snapshots();
         for (SnapshotId snapshotId : snapshotIds) {
-            snapshotSet.add(repository.readSnapshot(snapshotId));
+            try {
+                snapshotSet.add(repository.readSnapshot(snapshotId));
+            } catch (Exception ex) {
+                if (ignoreUnavailable) {
+                    logger.warn("failed to get snapshot : " + snapshotId, ex);
```

Comment:
please don't use String concatenation in logger calls, but positional parameters, for example `logger.warn("failed to get snapshot [{}]", ex, snapshotId);`